crates/milli/src/update/new/extract/faceted/extract_facets.rs

```diff
@@ -90,17 +90,11 @@ impl FacetedDocidsExtractor {
         let mut cached_sorter = context.data.borrow_mut_or_yield();
         let mut del_add_facet_value = DelAddFacetValue::new(&context.doc_alloc);
         let docid = document_change.docid();
-        let res = match document_change {
-            DocumentChange::Deletion(inner) => extract_document_facets(
-                inner.current(rtxn, index, context.db_fields_ids_map)?,
-                inner.external_document_id(),
-                new_fields_ids_map.deref_mut(),
-                filterable_attributes,
-                sortable_fields,
-                asc_desc_fields,
-                distinct_field,
-                is_geo_enabled,
-                &mut |fid, meta, depth, value| {
+
+        // Macro expanding to an insertion/deletion facet fn
+        macro_rules! facet_fn {
```

Comment:
Can you explain why the macro is necessary here, please?

```suggestion
        // Macro expanding to an insertion/deletion facet fn,
        // using a macro avoid to borrow the parameters as mutable in both closures at the same time by postponing their creation 
        macro_rules! facet_fn {
```